### PR TITLE
ContainerClient: Easily make HTTP requests against a container

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: seaworthy/client.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if self._session is not None:
+            self._session.close()
+            self._session = None
+
+    def __exit__(self, *args):
+        self.close()
+
+    def _get_session(self):
+        if self._session is None:
+            self._session = self._session_factory()
```
Comment (JayH5) [inline seaworthy/client.py]: I wasn't entirely sure where to create these sessions... With requests, you can still make requests with a session object that has been closed (I assume it is reopened or something?).

Comment (jerith) [reply seaworthy/client.py]: I think just create the session in `__init__()` and close it in `close()` (which is called from `__exit__()`. There doesn't seem to be any real cost to having an unused session lying around, and all closing the session does is close all the lurking connections and such.

File: seaworthy/tests-core/test_client.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        """
+        client = ContainerClient('127.0.0.1', '12345')
+
+        responses.add(responses.GET, 'http://127.0.0.1:12345/', status=200)
+        response = client.request('GET', [])
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(len(responses.calls), 1)
+        [call] = responses.calls
```
Comment (jerith) [inline seaworthy/tests-core/test_client.py]: I think you can drop the `len(response.calls)` check. `[call] = responses.calls` will give you all the same information for a failure.

File: seaworthy/tests-core/test_client.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        self.assertEqual(call.request.url, 'http://127.0.0.1:12345/')
+
+    @responses.activate
+    def test_url_defaults(self):
+        """
+        When the container client is configured with a host address and port,
+        and some URL defaults are set, requests are made agains that address
```
Comment (jerith) [inline seaworthy/tests-core/test_client.py]: "agains"? :-P

I think "[...] requests are made *to* that address [...]" would be better anyway.

**Pull request merged**